src/coreclr/src/vm/clrex.h

```diff
@@ -22,14 +22,23 @@ class AssemblySpec;
 class PEFile;
 class PEAssembly;
 
+enum StackTraceElementFlags
+{
+    // Set if this element represents the last frame of the foreign exception stack trace
+    STEF_LAST_FRAME_FROM_FOREIGN_STACK_TRACE = 0x0001,
+
+    // Set if the "ip" field has already been adjusted (decremented)
+    STEF_IP_ADJUSTED = 0x0002,
+};
+
+// This struct is used by SOS in the diagnostic repo.
+// See: https://github.com/dotnet/diagnostics/blob/master/src/SOS/Strike/strike.cpp#L2245
 struct StackTraceElement
 {
     UINT_PTR        ip;
     UINT_PTR        sp;
     PTR_MethodDesc  pFunc;
-    // TRUE if this element represents the last frame of the foreign
-    // exception stack trace.
-    BOOL			fIsLastFrameFromForeignStackTrace;
+    INT             flags;      // This is StackTraceElementFlags but it needs to be "int" sized for compatibility with SOS.
```

Comment:
Nice! This looks like a much more elegant solution to handle the SOS back-compat issues : )

Somewhat of a tangent, but we should also look at what we can do in SOS to warn users when the version they are using is too old to work properly with the runtime they are trying to debug. Best case is we never need to make a breaking change, but if we ever did need to, a little groundwork now could really payoff.

Reply:
I'm thinking that SOS can use the runtime version (i.e. 5.0.20.7701) for things like this since the major/minor version is the actual product version of the runtime.

Reply:
Lets chat offline on this part, I'm not sure what you are proposing just yet : )